ledger-api/grpc-definitions/com/daml/ledger/api/v1/experimental_features.proto

```diff
@@ -16,7 +16,7 @@ option csharp_namespace = "Com.Daml.Ledger.Api.V1";
 
 // See the feature message definitions for descriptions.
 message ExperimentalFeatures {
-  ExperimentalSelfServiceErrorCodes self_service_error_codes = 1;
+  ExperimentalSelfServiceErrorCodes self_service_error_codes = 1 [deprecated=true];
```

Comment:
```suggestion
  reserved 1; // was  self_service_error_codes
```
No need to carry it forward

Reply:
Don't we need to keep it here not to break backward compatibility? AFAIK it's been already released in 1.18.

Reply:
yes, you are right, the problem is with the clients written using sdk 1.18 for instance `ledger-api-test-tool`. It will ask for feature descriptors and based on that check lagacy or post 1.18 error codes. If it doesn't find the self_service_error_codes field, it will assume it's an old ledger and will check legacy codes.

Reply:
We can break compatibility in experimental_features as described in this file.
>  IMPORTANT: in contrast to other parts of the Ledger API, only json-wire backwards
            compatibility guarantees are given for the messages in this file.
        
My understanding is, if we want this test-tool to run against older servers we probably need to leave this field. ..but then should it still support asserting on legacy error codes? (meaning we cannot weed out legacy checks from it?) or just fail fast saying legacy error checking is not supported?

If we won't run this tool against older servers we can mark this field as `reserved`.

Reply:
Adjusted compatibility tests configuration: d423f5bc4a8ece48ecb376f11e58adf2af120f1e